### PR TITLE
chain: Request data message handler test

### LINKED ISSUE #1698: MsgRequest handler test

### PR DESCRIPTION
**Pull request by perimeko**
Fixed #1698

### CODE REVIEW

File: chain/x/oracle/handler_test.go [hunk 1]
```diff
@@ -662,3 +663,81 @@ func TestReportFailed(t *testing.T) {
...
+
+	events := result.GetEvents()
+
+	// Request Message Events
+	require.Equal(t, 4, len(events))
+	require.Equal(t, types.EventTypeRequest, events[0].Type)
+	require.Equal(t, types.AttributeKeyID, string(events[0].Attributes[0].Key))
+	require.Equal(t, types.AttributeKeyValidator, string(events[0].Attributes[1].Key))
+
+	for reqID := 1; reqID <= 3; reqID++ {
```
Comment (taobun) [inline chain/x/oracle/handler_test.go]: Use i is better reqID that misleading to request id but it doesn't

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — DISMISSED** [review]

**Comment by sorawit** [comment]
@perimeko Please resolve conflict. Likely a small conflict with #1793

**Review by taobun — DISMISSED** [review]

**Review by Benzbeeb — DISMISSED** [review]

**Review by Benzbeeb — DISMISSED** [review]
💯

**Review by taobun — APPROVED** [review]

**Review by Benzbeeb — APPROVED** [review]
⛓️ 💯

**Review by sorawit — APPROVED** [review]

**Pull request merged**